kedro/runner/parallel_runner.py

```diff
@@ -244,6 +199,16 @@ def _validate_catalog(cls, catalog: DataCatalog, pipeline: Pipeline):
                 f"MemoryDatasets"
             )
 
+    def _set_manager_datasets(self, catalog, pipeline):
+        for dataset in pipeline.datasets():
+            try:
+                catalog._get_dataset(dataset)
+            except DatasetNotFoundError:
+                pass
```

Comment:
`if dataset in catalog` wouldn't catch any datasets that match a pattern, because those are only materialised at access (load/save/exists). But I'll improve the rest.